src/lib/demux.rs

```diff
@@ -38,6 +38,12 @@ impl ExtractedBarcode {
         Self::default()
     }
 
+    /// Create a new [`ExtractedBarcode`] from the delimited barcode, with `+` as the delimiter.
+    fn from_delimited(delimited: &[u8]) -> Self {
+        let concatenated = delimited.splitn(2, |c| *c == b'+').collect_vec().concat();
```

Comment:
What's the reason to limit it to two?
Also if this is invoked for every read, it would be nice to find a way to do this without creating an intermediate Vec.

Reply:
I am assuming single or dual indexed sample barcoded runs.  I am not expecting triple-barcode or higher, so this should be faster